playbooks/ooi.yaml

```diff
@@ -0,0 +1,146 @@
+        - name: Add noauth2 to OCCI API composite to nova api-paste.ini
+          ini_file:
+            path: /etc/nova/api-paste.ini
+            section: composite:occi_api_11
+            option: noauth2
+            # yamllint disable-line rule:line-length
+            value: cors http_proxy_to_wsgi compute_req_id faultwrap sizelimit osprofiler noauth2 occi osapi_compute_app_v21
+          notify:
+            - restart nova-api
+          tags:
+            - skip_ansible_lint
+        - name: Add keystone to OCCI API composite to nova api-paste.ini
+          ini_file:
+            path: /etc/nova/api-paste.ini
+            section: composite:occi_api_11
+            option: keystone
+            # yamllint disable-line rule:line-length
+            value: cors http_proxy_to_wsgi compute_req_id faultwrap sizelimit osprofiler authtoken keystonecontext legacy_v2_compatible occi osapi_compute_app_v21
+          notify:
+            - restart nova-api
+          tags:
+            - skip_ansible_lint
+        - name: Add use to OOI composite to nova api-paste.ini
+          ini_file:
+            path: /etc/nova/api-paste.ini
+            section: composite:ooi
+            option: use
+            value: call:nova.api.openstack.urlmap:urlmap_factory
+          notify:
+            - restart nova-api
+        - name: Add occi1.1 to OOI composite to nova api-paste.ini
+          ini_file:
+            path: /etc/nova/api-paste.ini
+            section: composite:ooi
+            option: /occi1.1
+            value: occi_api_11
+          notify:
+            - restart nova-api
+
+    - name: Configure OOI in nova.conf
+      block:
+        - name: Enable OOI API in nova.conf
+          ini_file:
+            path: /etc/nova/nova.conf
+            section: DEFAULT
+            option: enabled_apis
+            value: osapi_compute,metadata,ooi
+            backup: yes
+          notify:
+            - restart nova-compute
+        - name: Configure OOI port in nova.conf
+          ini_file:
+            path: /etc/nova/nova.conf
+            section: DEFAULT
+            option: ooi_listen_port
+            value: 8787
+            backup: yes
+          notify:
+            - restart nova-compute
+
+    - name: Create OCCI endpoint
+      block:
+        - name: Create OCCI service
+          os_keystone_service:
+            cloud: "{{ server_fqdn }}"
+            state: present
+            name: occi
+            service_type: occi
+            description: OCCI interface
+        # XXX To be replaced by an HTTPS endpoint
+        - name: Create OCCI endpoint for OOI
+          os_keystone_endpoint:
+            cloud: "{{ server_fqdn }}"
+            service: occi
+            endpoint_interface: public
+            url: http://{{ server_fqdn }}:8787/occi1.1
+            region: RegionOne
+            state: present
+      delegate_to: localhost
+      become: no
+
+    - name: Open OCCI port in iptables
+      lineinfile:
```

Comment:
Have you tried using http://docs.ansible.com/ansible/latest/modules/iptables_module.html ?